### DISCUSSION TITLE
Native iOS and Android Apps

### DISCUSSION THREAD

**Discussion by abose** [phcode-dev/phoenix#1046] [General]
Please upvote for Android and iOS apps of phcode.dev. 

We'd love to hear your ideas! Feel free to share any feature requests or suggestions for our native apps. Your input is truly appreciated!

Desktop native apps: https://github.com/phcode-dev/phoenix/discussions/1047

**Comment by thasch9bl** [comment]
Estou aprendendo programação e nas minhas pesquisas para aprender cheguei até aqui, porém não tenho outro dispositivo de acesso apenas o celular.
Que nada que deu😔
***https://github.com/thasch9bl***

**Reply by abose** [reply]
@thasch9bl Thanks for the feedback. Do you have any suggestions for us to improve your learning experience on mobile devices? What features would you like that would make a code editor usable on mobile devices?

Would you consider using an external keyboard with your mobile device for coding?

**Reply by abose** [reply]
@thasch9bl Obrigado pelo feedback. Você tem alguma sugestão para melhorarmos sua experiência de aprendizado em dispositivos móveis? Quais recursos você gostaria que tornassem um editor de código utilizável em dispositivos móveis?

Você consideraria usar um teclado externo com seu dispositivo móvel para codificação?
- traduzido pelo Google

**Comment by Kowshik-Eggoni** [comment]
Any plans to support cross platform tools like flutter, and react native? That would be amazing to see!

**Reply by abose** [reply]
It is not currently being tracked. Please create a discussion And add any requirements for us to investigate.

**Reply by samolukadjo** [reply]
> It is not currently being tracked. Please create a discussion And add any requirements for us to investigate.

On android it's technically possible to set up a development environment with node etc. and it's packages like for React Native or pretty much any of the other framework. The process requires Acode editor, a specific extension for it, termux, and keeping all your project files in Termuxes storage. It would be great if ph could provide a simpler and more straightforward experience that's less finicky.

**Comment by charles-prof** [comment]
If you set a counter to upvote, it will be better.

Counter may be placed in the website with a nice landing page with mobile (Android/iOS) mockup screenshots

**Comment by brendancarr** [comment]
Have been trying out the web based version in a mobile environment and it is already much better than other syntax-based code editors. I would love a built in ssh/ftp browser with secure local storage to save session data. I use RaiDrive to connect to all of my servers on a desktop, negating the need, but having it easily accessible anywhere would be $$$. Currently working with phpseclib for other applications, it isn't a stretch to use that to display a file tree and load files on the fly. Or SSH2 for node ? I have ideas.

**Reply by abose** [reply]
phcode works the same in web and mobile browsers. The only issue is that the GUI is not small screen friendly.

But we will be rolling out full support for safari, iOS and android tablets in the browser in July. (Note that these are not native apps, but from tablet browser).  Small screens and usage without a keyboard or mouse will still be not the priority for July release.

The next release will also most likely have full featured Git support in the browser- that works fully locally in the browser. Terminal support is scheduled after that.

Reg ssh/ftp, i'd like to know more about the workflows you would want inside phcode.

**Reply by brendancarr** [reply]
I use it like the old OSX Coda, directly editing on the server. In Coda the SSH/SFTP/FTP was attached to the project, when you save a file you save it directly on the server or you have a mirror locally that pushes up to the server upon a file change, or when you hit a sync/publish/commit button. Out of those three options I use the first, the second would be great with a Google Drive/S3 bucket to host the 'local' copy, and it would essentially mean I can jump on any computer, log in, pull up a project, and edit it on the fly from any device anywhere. You could tie it to a repo even, and commit after testing changes on the live server. Any device becomes a headless IDE.

If it were me, I'd integrate the phcode frontend into our admin dashboard, which links to our whm servers, tracks billing, etc. that way I can log into a single location and have access to everything needed to run the business. The open project screen would pull from the client list. That way it's a self hosted copy that doesn't require your team to maintain hosting somewhere for us to use it. All login information would be stored in a central location managed by us and encrypted with salt or 2fa, or both.

**Reply by abose** [reply]
This sounds similar to what https://git-ftp.github.io/ is for. 

Do you mean from the browser ? For browsers, a backend will be needed to do any kind of non http related connections. And if you are on a private network, it has to be hosted within the network too. From a desktop app its doable though.

**Comment by BohBanny** [comment]
- [ ]

**Comment by kevinrook** [comment]
I really love using phcode on mac and web browser, but so desperately need an iPadOS app for my iPad Pro. The browser version dosnt like the ipad and things like local storage dont work which is a pain.

Any news on when there will be a native app?

**Reply by abose** [reply]
1. The browser version dosnt like the ipad - can you give more context on what things are broken?
2. Also would be helpful in understanding the workflow with ipad, what features are needed. For eg: do you need to run node or other frameworks, or just html editing would do?
